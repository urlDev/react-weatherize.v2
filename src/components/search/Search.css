.search {
    width: 90%;
    margin: auto;
    margin-top: 20px;
}

form {
    display: flex;
}

.input {
    height: 100%;
    width: 80%;
    background: none;
    border: none;
    outline: none;
    border-bottom: 1px solid white;
    font-family: 'Roboto', sans-serif;
    margin-right: auto;
    margin-left: auto;
    margin-top: auto;
    color: white;
    font-size: 18px;
}

input::placeholder {
    color: white;
}

button {
    border: none;
    padding: 0;
    width: auto;
    height: auto;
}

button img {
    background: #d46b05;
    margin: 0;
    width: 10px;
    padding: 15px;
    display: flex;
}

@media (min-width: 900px) {
    .search {
        width: 100%;
        margin: 0;
    }
    .input {
        width: calc(80% - 20px - 60px);
    }
    button img {
        padding: 30px;
        width: 20px;
    }
}