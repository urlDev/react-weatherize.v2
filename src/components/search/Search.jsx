/* eslint-disable operator-linebreak */
/* eslint-disable no-unused-expressions */
import React from 'react';
import { useDispatch } from '../../tests/__mocks__/reactReduxHooks';
import { fetchWeather } from '../../redux/weather/weatherActions';

import './Search.css';

const Search = () => {
  const [input, setInput] = React.useState('');
  const dispatch = useDispatch();

  const handleChange = (e) => {
    setInput(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(fetchWeather(input));
    setInput('');
  };

  return (
    <div className="search">
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          className="input"
          placeholder="Previous cities"
          value={input}
          onChange={handleChange}
          required
        />
        <button type="submit">
          <img
            // eslint-disable-next-line global-require
            src={require('../../assets/search.svg')}
            alt="search magnifier"
          />
        </button>
      </form>
    </div>
  );
};

export default Search;
