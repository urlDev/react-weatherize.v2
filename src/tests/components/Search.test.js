/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';

import Search from '../../components/search/Search';

import { mockStore } from '../__mocks__/store';
import * as ReactReduxHooks from '../__mocks__/reactReduxHooks';

describe('Search component', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore({});
    jest
      .spyOn(ReactReduxHooks, 'useDispatch')
      .mockImplementation(() => store.dispatch);
  });

  test('Should render the component correctly', () => {
    wrapper = shallow(<Search store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render form with input value', () => {
    const value = 'Helsinki';
    wrapper = shallow(<Search value={value} />);
    wrapper.find('input').simulate('change', { target: { value } });
    expect(wrapper).toMatchSnapshot();
  });
  test('Should submit form with valid input', async () => {
    const onSubmitSpy = jest.fn();
    const value = 'Helsinki';

    wrapper = shallow(
      <Search onSubmit={onSubmitSpy} value={value} store={store} />,
    );

    wrapper.find('form').simulate('submit', {
      preventDefault: () => {},
    });
    onSubmitSpy();
    expect(onSubmitSpy).toHaveBeenCalled();
  });
});
