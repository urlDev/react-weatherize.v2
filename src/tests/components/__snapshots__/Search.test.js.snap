// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component Should render form with input value 1`] = `
<div
  className="search"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Previous cities"
      required={true}
      type="text"
      value="Helsinki"
    />
    <button
      type="submit"
    >
      <img
        alt="search magnifier"
        src={Object {}}
      />
    </button>
  </form>
</div>
`;

exports[`Search component Should render the component correctly 1`] = `
<div
  className="search"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Previous cities"
      required={true}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      <img
        alt="search magnifier"
        src={Object {}}
      />
    </button>
  </form>
</div>
`;
